index.bs

```diff
@@ -2152,6 +2157,25 @@ browsingContext.ReadinessState = "none" / "interactive" / "complete"
 The <code>browsingContext.ReadinessState</code> type represents the stage of
 document loading at which a navigation command will return.
 
+#### The browsingContext.Cookie Type #### {#type-browsingContext-Cookie}
```

Comment:
we probably want cookies outside of the browsing context module (except for the command to get a partition key or other commands that require the context as input)? I am thinking a new domain called `storage` would be good?  cc @jgraham

Reply:
That sounds fine to me. Currently I haven't included a command to get a cookie's partition key, are you thinking something like `broswsingContext.getCookiePartitionKey` for that?

Reply:
Yes to a `storage` domain.

For the partition key, I think one question is whether it's sufficient to be able to get the key only for already loaded contexts. That is roughly equivalent to not exposing the key at all, but only being able to set cookies for a currently loaded page, because you always need to load a page, get the key, and then use that to set cookies.

I slightly wondered if we should define some partition key fields and have something in the capabilities response that returns the list of keys recognised by the current browser. So we might define `sourceOrigin` as a key. Then a browser with only SOP would just return an empty key set like:
```js
storagePartitionKeys: {}
```
A browser that required keying on `sourceOrigin` and couldn't provide any defaults would return something like:
```js
storagePartitionKeys: {sourceOrigin: {required: true}}
```
And then something like current gecko with containers could provide a custom key like `moz:userContext` like so:
```js
storagePartitionKeys: {sourceOrigin: {required: true}, "moz:userContext": {default: null}}
```
So then clients/users could know that to set storage properties you'd _have_ to provide a value for the `sourceOrigin` key and that `moz:userContext` is another part of the key, but if you don't provide it you'll get some default value that hopefully does the right thing for most cases. This doesn't totally solve the forwards compatibility problem, but at least makes it possible for clients to implement a good UX (by telling you exactly what you need to change if there's a change, and by allowing users to provide a set of properties that's a superset of what the client expects).

Reply:
I just want to make a clarification here, when we're talking about partition key for cookies, are we talking about partitioned cookies as defined in [CHIPS](https://developer.mozilla.org/en-US/docs/Web/Privacy/Partitioned_cookies) or are we talking about cookies defined within Mozilla's containers for different user contexts?